addons/udes_stock/models/stock_move.py

```diff
@@ -443,9 +454,13 @@ def refactor_action_group_by_move_line_key(self):
                 else:
                     group_moves |= move
 
-            Picking._new_picking_for_group(key, group_moves)
+            origin = None
+            if len(group_pickings) == 1:
+                origin = group_pickings.origin
+            Picking._new_picking_for_group(key, group_moves, origin=origin)
             result_moves |= group_moves
 
+        # NOTE: empty pick = no moves and no move_lines
```

Comment:
I don't think this comment is necessary, the filtered should be self explanatory.

Reply:
The filtered is wrong, we should fix it.

Reply:
Just ignore my comment misread move_lines by move_line_ids